ApiDocs.Validation/MultipartMime/MultipartMimeContent.cs

```diff
@@ -0,0 +1,219 @@
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE 
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+
+namespace ApiDocs.Validation.MultipartMime
+{
+    using Tags;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+    using System.IO;
+
+    public class MultipartMimeContent
+    {
+        /// <summary>
+        /// Create a new multipart mime document
+        /// </summary>
+        public MultipartMimeContent()
+        {
+            ContentType = new MimeContentType("multipart/related");
+            Boundary = Guid.NewGuid().ToString("d");
+            Parts = new List<MessagePart>();
+        }
+
+        /// <summary>
+        /// Parse an existing multipart mime body into parts
+        /// </summary>
+        /// <param name="contentType"></param>
+        /// <param name="body"></param>
+        public MultipartMimeContent(string contentType, string body)
+        {
+            ContentType = new MimeContentType(contentType);
+            Parts = ParseBodyIntoParts(body, Boundary);
+        }
+
+        public MultipartMimeContent(MimeContentType contentType, string body)
+        {
+            ContentType = contentType;
+            Parts = ParseBodyIntoParts(body, Boundary);
+        }
+
+        /// <summary>
+        /// The boundary identifier for the multipart message
+        /// </summary>
+        public string Boundary {
+            get { return ContentType.GetArgumentValue("boundary").TextBetweenCharacters('"', '"'); }
+            set { ContentType.SetArgumentValue("boundary", $"\"{value}\""); }
+        }
+
+        /// <summary>
+        /// The full value of the content-type header for this message
+        /// </summary>
+        public MimeContentType ContentType { get; set; }
+
+        /// <summary>
+        /// The actual MIME type for the multipart mime message, usually multipart/related
+        /// </summary>
+        public string MultipartType
+        {
+            get { return ContentType.MimeType; }
+            set { ContentType.MimeType = value; }
+        }
+
+
+        public MessagePart PartWithId(string id)
+        {
+            return (from p in Parts where p.Id == id select p).FirstOrDefault();
+        }
+
+        public IEnumerable<MessagePart> PartsWithContentType(string contentType)
+        {
+            return (from p in Parts where p.ContentType.MimeType == contentType select p);
+        }
+
+        private static List<MessagePart> ParseBodyIntoParts(string body, string boundary)
+        {
+            var parts = new List<MessagePart>();
+
+            StringReader reader = new StringReader(body);
+            string nextLine = reader.ReadLine();
+            StringBuilder partbody = new StringBuilder();
+            while(nextLine != null)
+            {
+                if (nextLine.Equals($"--{boundary}"))
```

Comment:
There probably was, if I we used something other than multipart/related. However finding one that supported that took longer than writing one quickly myself.